File: eBarbershop.Model/Rezervacija.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarbershop.Model
{
    public class Rezervacija
{
    public int RezervacijaId { get; set; }

    public DateTime DatumRezervacije { get; set; }

    public int KorisnikId { get; set; }
    public Korisnik Korisnik { get; set; }

    public int KlijentId { get; set; }
    public Korisnik Klijent { get; set; }

    public int UslugaId { get; set; }
    public Usluga Usluga { get; set; }

}

}

File: UI/ebarbershop_mobile/lib/models/rezervacija.dart
import 'package:ebarbershop_mobile/models/korisnik.dart';
import 'package:ebarbershop_mobile/models/termin.dart';
import 'package:ebarbershop_mobile/models/usluga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int? rezervacijaId;
  DateTime? datumRezervacije;
  int? korisnikId; // frizer
  Korisnik? korisnik;
  int? klijentId; // logirani korisnik
  Korisnik? klijent;
  int? uslugaId;
  Usluga? usluga;

  Rezervacija({
    this.rezervacijaId,
    this.datumRezervacije,
    this.korisnikId,
    this.korisnik,
    this.klijentId,
    this.klijent,
    this.uslugaId,
    this.usluga,
  });

  factory Rezervacija.fromJson(Map<String, dynamic> json) => _$RezervacijaFromJson(json);
  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}